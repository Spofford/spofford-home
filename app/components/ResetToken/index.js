import React from "react"
import cssModules from "react-css-modules"
import style from "./style.css"
import { connect } from "react-redux"
import { reset } from "../../redux/actions"
import { Redirect } from 'react-router-dom'
import { default as Header } from "../Header"

export class ResetToken extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      token: "",
      redirect: false
    }

    this.submit = this.submit.bind(this)

  }

  submit() {
    const reset = {
      password: document.getElementById("reset-password").value,
      token: this.state.token
    }

    this.props.reset(reset)
  }

  componentDidMount = () => {
    this.setState({
      token: this.props.location.search.split('=')[1]
    })
  }

  componentDidUpdate(prevProps) {
    if (prevProps.user.id != this.props.user.id) {
      this.setState({
        redirect: true
      })
    }
  }

  render() {
    if (this.state.redirect) {
      return <Redirect to='/submissions'/>;
    }
      return (
        <div className="login">
          <Header />
          <div className="body-container">
            <div className="copy-container">
              <div className="wrapper">
                <h2>Password Reset</h2>
                <div className="form">
                  <div className="inputGroup">
                    <label>New Password
                      <input
                      className="input"
                      type="text"
                      autoComplete="password"
                      id="reset-password" />
                      </label>
                  </div>
                  <button className="green" onClick={this.submit}>Submit</button>
                </div>
              </div>
            </div>
          </div>
        </div>
      )
    }

}

const mapDispatchToProps = {
  reset: reset
};

const mapStateToProps = state => ({
  user: state.user

})

export default connect(mapStateToProps, mapDispatchToProps)(cssModules(ResetToken, style))
