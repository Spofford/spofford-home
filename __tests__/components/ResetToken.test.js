import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import ResetToken from '../../app/components/ResetToken';


const mockStore = configureStore();
const initialState = {
  user: {
    email: "amhasler@gmail.com",
    first_name: "Adam",
    last_name: "Hasler",
    role: 1,
    id: 20,
    charges: []
  }
}
const store = mockStore(initialState);

describe('<ResteToken />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<ResetToken store={store} />);
      // const component = wrapper.dive();

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    // test to render model from contentful
  });
});
