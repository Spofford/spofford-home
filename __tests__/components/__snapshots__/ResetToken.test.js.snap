// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResteToken /> render() renders the component 1`] = `
<ResetToken
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={
    Object {
      "charges": Array [],
      "email": "amhasler@gmail.com",
      "first_name": "Adam",
      "id": 20,
      "last_name": "Hasler",
      "role": 1,
    }
  }
/>
`;
